docs/gen_reference_doc.py

```diff
@@ -35,690 +36,784 @@
-	'bencode.hpp': 'Bencoding',
-	'lazy_entry.hpp': 'Bencoding',
-	'bdecode.hpp': 'Bdecoding',
-	'entry.hpp': 'Bencoding',
-	'time.hpp': 'Time',
-	'escape_string.hpp': 'Utility',
-	'enum_net.hpp': 'Network',
-	'broadcast_socket.hpp': 'Network',
-	'socket.hpp': 'Network',
-	'socket_io.hpp': 'Network',
-	'bitfield.hpp': 'Utility',
-	'sha1_hash.hpp': 'Utility',
-	'hasher.hpp': 'Utility',
-	'hasher512.hpp': 'Utility',
-	'identify_client.hpp': 'Utility',
-	'ip_filter.hpp': 'Filter',
-	'session_settings.hpp': 'Settings',
-	'settings_pack.hpp': 'Settings',
-	'operations.hpp': 'Alerts',
-	'disk_buffer_holder.hpp': 'Custom Storage',
-	'alert_dispatcher.hpp': 'Alerts',
+    'ed25519.hpp': 'ed25519',
+    'session.hpp': 'Core',
+    'add_torrent_params.hpp': 'Core',
+    'session_status.hpp': 'Core',
+    'error_code.hpp': 'Error Codes',
+    'storage.hpp': 'Custom Storage',
+    'storage_defs.hpp': 'Storage',
+    'file_storage.hpp': 'Storage',
+    'file_pool.hpp': 'Custom Storage',
+    'extensions.hpp': 'Plugins',
+    'ut_metadata.hpp': 'Plugins',
+    'ut_pex.hpp': 'Plugins',
+    'ut_trackers.hpp': 'Plugins',
+    'smart_ban.hpp': 'Plugins',
+    'create_torrent.hpp': 'Create Torrents',
+    'alert.hpp': 'Alerts',
+    'alert_types.hpp': 'Alerts',
+    'bencode.hpp': 'Bencoding',
+    'lazy_entry.hpp': 'Bencoding',
+    'bdecode.hpp': 'Bdecoding',
+    'entry.hpp': 'Bencoding',
+    'time.hpp': 'Time',
+    'escape_string.hpp': 'Utility',
+    'enum_net.hpp': 'Network',
+    'broadcast_socket.hpp': 'Network',
+    'socket.hpp': 'Network',
+    'socket_io.hpp': 'Network',
+    'bitfield.hpp': 'Utility',
+    'sha1_hash.hpp': 'Utility',
+    'hasher.hpp': 'Utility',
+    'hasher512.hpp': 'Utility',
+    'identify_client.hpp': 'Utility',
+    'ip_filter.hpp': 'Filter',
+    'session_settings.hpp': 'Settings',
+    'settings_pack.hpp': 'Settings',
+    'operations.hpp': 'Alerts',
+    'disk_buffer_holder.hpp': 'Custom Storage',
+    'alert_dispatcher.hpp': 'Alerts',
 }
 
 category_fun_mapping = {
-	'min_memory_usage()': 'Settings',
-	'high_performance_seed()': 'Settings',
-	'cache_status': 'Core',
+    'min_memory_usage()': 'Settings',
+    'high_performance_seed()': 'Settings',
+    'cache_status': 'Core',
 }
 
+
 def categorize_symbol(name, filename):
-	f = os.path.split(filename)[1]
+    f = os.path.split(filename)[1]
+
+    if name.endswith('_category()') \
+            or name.endswith('_error_code') \
+            or name.endswith('error_code_enum'):
+        return 'Error Codes'
 
-	if name.endswith('_category()') \
-		or name.endswith('_error_code') \
-		or name.endswith('error_code_enum'):
-		return 'Error Codes'
+    if name in category_fun_mapping:
+        return category_fun_mapping[name]
 
-	if name in category_fun_mapping:
-		return category_fun_mapping[name]
+    if f in category_mapping:
+        return category_mapping[f]
 
-	if f in category_mapping:
-		return category_mapping[f]
+    return 'Core'
 
-	return 'Core'
 
```

Comment:
this additional blank line, does it serve a purpose?

Reply:
There should be 2 blank lines between top level classes and functions.

https://lintlyci.github.io/Flake8Rules/rules/E302.html